File: app/src/main/java/de/bitb/astroskop/utils/VersionCodeUtils.java
package de.bornholdtlee.dbsystel.utils;


import android.content.Context;
import android.content.pm.PackageManager;
import android.view.View;
import android.widget.TextView;

import de.bornholdtlee.dbsystel.BuildConfig;
import de.bornholdtlee.dbsystel.Constants;
import de.bornholdtlee.dbsystel.R;
import de.bornholdtlee.dbsystel.helper.Logger;

public final class VersionCodeUtils {

    private VersionCodeUtils() {
        super();
    }

    public static void checkVersionAndDisplayVersionCode(Context context, TextView versionCode) {
        final String buildVariant = BuildConfig.FLAVOR + (BuildConfig.DEBUG ? Constants.BUILD_TYPE_DEBUG : Constants.BUILD_TYPE_RELEASE);
        versionCode.setVisibility(View.VISIBLE);
        versionCode.setText(String.format(context.getString(R.string.version_code), buildVariant, getVersionName(context), getVersionCode(context)));
    }

    private static String getVersionName(Context context) {
        try {
            return context.getPackageManager().getPackageInfo(context.getPackageName(), 0).versionName;
        } catch (PackageManager.NameNotFoundException e) {
            Logger.error(e.getMessage());
        }
        return "";
    }

    private static int getVersionCode(Context context) {
        try {
            return context.getPackageManager().getPackageInfo(context.getPackageName(), 0).versionCode;
        } catch (PackageManager.NameNotFoundException e) {
            Logger.error(e.getMessage());
        }
        return 0;
    }

}

File: app/src/main/java/de/bitb/astroskop/utils/VersionCodeUtils.kt
package de.bitb.astroskop.utils


import android.content.Context
import android.content.pm.PackageManager
import android.view.View
import android.widget.TextView

import de.bitb.astroskop.BuildConfig
import de.bitb.astroskop.Constants
import de.bitb.astroskop.helper.Logger

class VersionCodeUtils private constructor() {
    companion object {

        fun checkVersionAndDisplayVersionCode(context: Context, versionCode: TextView) {
            val buildVariant = BuildConfig.FLAVOR + if (BuildConfig.DEBUG) Constants.BUILD_TYPE_DEBUG else Constants.BUILD_TYPE_RELEASE
            versionCode.visibility = View.VISIBLE
            //        versionCode.setText(String.format(context.getString(R.string.version_code), buildVariant, getVersionName(context), getVersionCode(context)));
        }

        private fun getVersionName(context: Context): String {
            try {
                return context.packageManager.getPackageInfo(context.packageName, 0).versionName
            } catch (e: PackageManager.NameNotFoundException) {
                Logger.error(e.message)
            }

            return ""
        }

        private fun getVersionCode(context: Context): Int {
            try {
                return context.packageManager.getPackageInfo(context.packageName, 0).versionCode
            } catch (e: PackageManager.NameNotFoundException) {
                Logger.error(e.message)
            }

            return 0
        }
    }

}
